apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/Playground/DocumentParams/Input/index.tsx

```diff
@@ -72,3 +95,232 @@ export function InputParams({
+  input,
+  setInput,
+  disabled = false,
+}: {
+  parameter: string
+  input: PlaygroundInput<InputSource>
+  setInput: (param: string, value: PlaygroundInput<InputSource>) => void
+  disabled?: boolean
+}) {
+  const { uploadFile, convertFile, isLoading } = useFiles()
+
+  const onTextChange = useCallback(
+    (e: ChangeEvent<HTMLTextAreaElement>) => {
+      setInput(parameter, { ...input, value: e.target.value })
+    },
+    [setInput],
+  )
+
+  const onDocumentChange = useCallback(
+    async (files: FileList | null) => {
+      const file = files?.[0]
+      if (file) {
+        const content = await convertFile({ file })
+        if (content) {
+          setInput(parameter, {
+            ...input,
+            value: content,
+            metadata: {
+              ...input.metadata,
+              filename: file.name,
+            },
+          })
+          return
+        }
+      }
+
+      setInput(parameter, {
+        ...input,
+        value: '',
+        metadata: {
+          ...input.metadata,
+          filename: undefined,
+        },
+      })
+    },
+    [convertFile, setInput],
+  )
+
+  const onImageChange = useCallback(
+    async (files: FileList | null) => {
+      const file = files?.[0]
+      if (file) {
+        const url = await uploadFile({ file })
+        if (url) {
+          setInput(parameter, {
+            ...input,
+            value: url,
+            metadata: {
+              ...input.metadata,
+              filename: file.name,
+            },
+          })
+          return
+        }
+      }
+
+      setInput(parameter, {
+        ...input,
+        value: '',
+        metadata: {
+          ...input.metadata,
+          filename: undefined,
+        },
+      })
+    },
+    [uploadFile, setInput],
+  )
+
+  switch (input.metadata.type || ParameterType.Text) {
+    case ParameterType.Text:
+      return (
+        <TextArea
+          value={input.value ?? ''}
+          onChange={onTextChange}
+          minRows={1}
+          maxRows={6}
+          disabled={disabled}
+        />
+      )
+
+    case ParameterType.Document:
+      return isLoading ? (
+        <ParameterInputSkeleton />
+      ) : (
+        <DropzoneInput
+          icon='fileUp'
+          inputSize='small'
+          placeholder='Upload document'
+          defaultFilename={input.metadata.filename}
+          onChange={onDocumentChange}
+          accept={undefined}
+          multiple={false}
+          disabled={disabled}
+        />
+      )
+
+    case ParameterType.Image:
+      return isLoading ? (
+        <ParameterInputSkeleton />
+      ) : (
+        <DropzoneInput
+          icon='imageUp'
+          inputSize='small'
+          placeholder='Upload image'
+          defaultFilename={input.metadata.filename}
+          onChange={onImageChange}
+          accept={SUPPORTED_IMAGE_TYPES.join(',')}
+          multiple={false}
+          disabled={disabled}
+        />
+      )
+
+    default:
+      return null
```

Comment:
should we throw an error or is this an expected case?

Reply:
we could throw an error